### PR TITLE
octopus: qa/upgrade: disable update_features test_notify with older client as lockowner

### PR DESCRIPTION
**Pull request by ideepika**
with the recent support for async rbd operations from pacific+ when an
older client(non async support) goes on upgrade, and simultaneously
interacts with a newer client which expects the requests to be async,
experiences hang; considering the return code for request completion to
be acknowledgement for async request, which then keeps waiting for
another acknowledgement of request completion.

this if happens should be a rare only when lockowner is an old client
and should be deferred if compatibility issues arises.

tests including DISABLE_UPDATE_FEATURES: "1" environment variable should skip update_features test_notify check

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/suites/upgrade-clients/client-upgrade-octopus-pacific/octopus-client-x/rbd/3-workload/rbd_notification_tests.yaml [hunk 1]
```diff
@@ -8,25 +8,23 @@ tasks:
       env:
         RBD_FEATURES: "61"
   - workunit:
-      #The line below to change to 'pacific'
-      branch: master
+      branch: pacific
```
Comment (trociny) [inline qa/suites/upgrade-clients/client-upgrade-octopus-pacific/octopus-client-x/rbd/3-workload/rbd_notification_tests.yaml:11]: @ideepika I was expecting this PR would just set DISABLE_UPDATE_FEATURES variable to skip the test. But I see more changes like this one. Could you explain why they are here?

Comment (ideepika) [reply qa/suites/upgrade-clients/client-upgrade-octopus-pacific/octopus-client-x/rbd/3-workload/rbd_notification_tests.yaml]: removed 2nd commit, in favor of: https://github.com/ceph/ceph/pull/41513#issuecomment-849705557

**Review by yuriw — APPROVED** [review]

**Comment by yuriw** [comment]
http://pulpito.ceph.com/teuthology-2021-05-27_17:55:15-upgrade-clients:client-upgrade-octopus-pacific-pacific-distro-basic-smithi/

**Review by trociny — APPROVED** [review]

**Comment by trociny** [comment]
@yuriw It will not work until #41514 is merged and backported.

**Review by idryomov — CHANGES_REQUESTED** [review]
Pending the cherry-pick of #41514.

**Review by trociny — APPROVED** [review]

**Comment by ideepika** [comment]
passed test: https://pulpito.ceph.com/ideepika-2021-06-01_10:24:30-upgrade-clients:client-upgrade-octopus-pacific-wip-49592-upgrade-distro-basic-smithi

**Comment by idryomov** [comment]
Following up on the workunit branch mess note from #41511:

```
tasks:
- parallel:
  - workunit:
      branch: octopus
      clients:
        client.0:
          - rbd/notify_master.sh
      env:
        RBD_FEATURES: "61"
  - workunit:
      branch: pacific      <--------------
      clients:
        client.1:
          - rbd/notify_slave.sh
      env:
        RBD_FEATURES: "61"
        RBD_DISABLE_UPDATE_FEATURES: "1"
- print: "**** done rbd: old librbd -> new librbd"
- parallel:
  - workunit:
      branch: octopus         <--------------
      clients:
        client.0:
          - rbd/notify_slave.sh
      env:
        RBD_FEATURES: "61"
  - workunit:
      branch: pacific
      clients:
        client.1:
          - rbd/notify_master.sh
      env:
        RBD_FEATURES: "61"
- print: "**** done rbd: new librbd -> old librbd"
```

**Comment by ideepika** [comment]
> Following up on the workunit branch mess note from #41511:
> 
> ```
> tasks:
> - parallel:
>   - workunit:
>       branch: octopus
>       clients:
>         client.0:
>           - rbd/notify_master.sh
>       env:
>         RBD_FEATURES: "61"
>   - workunit:
>       branch: pacific      <--------------
>       clients:
>         client.1:
>           - rbd/notify_slave.sh
>       env:
>         RBD_FEATURES: "61"
>         RBD_DISABLE_UPDATE_FEATURES: "1"
> - print: "**** done rbd: old librbd -> new librbd"
> - parallel:
>   - workunit:
>       branch: octopus         <--------------
>       clients:
>         client.0:
>           - rbd/notify_slave.sh
>       env:
>         RBD_FEATURES: "61"
>   - workunit:
>       branch: pacific
>       clients:
>         client.1:
>           - rbd/notify_master.sh
>       env:
>         RBD_FEATURES: "61"
> - print: "**** done rbd: new librbd -> old librbd"
> ```
addressed branch comment for:
nautilus: https://github.com/ceph/ceph/pull/41513/commits/a564611f625a47723e373cdfa108c6dde5a31d24
octopus: https://github.com/ceph/ceph/pull/41511/commits/8b452ff8b104c580d96163b85cb6efef8aff1868
pacific backport: https://github.com/ceph/ceph/pull/41629

**Comment by idryomov** [comment]
I'd make it a single commit.

**Comment by ideepika** [comment]
> I'd make it a single commit.

sure, done!

**Comment by idryomov** [comment]
https://pulpito.ceph.com/dis-2021-06-01_21:08:01-upgrade-clients:client-upgrade-octopus-pacific-pacific-distro-basic-smithi/

**Review by idryomov — APPROVED** [review]

**Pull request merged**